core/google/cloud/iterator.py

```diff
@@ -393,6 +407,24 @@ def __init__(self, client, page_iter, item_to_value, max_results=None):
             max_results=max_results)
         self._page_iter = self._wrap_gax(page_iter)
 
+    def _next_page(self, page_iter):
+        """Get the next page in the iterator.
+
+        :type page_iter: :class:`~google.gax.PageIterator`
+        :param page_iter: The GAX page iterator to consume.
+
+        :rtype: :class:`Page`
+        :returns: The next page in the iterator (or :data:`None` if
+                  there are no pages left).
+        """
+        try:
+            items = six.next(page_iter)
+            page = Page(self, items, self._item_to_value)
+            self.next_page_token = page_iter.page_token or None
+            return page
+        except StopIteration:
+            return None
+
     def _wrap_gax(self, page_iter):
```

Comment:
Would it be better to rename `_wrap_gax` -> `_gax_page_iter`, to indicate its parallel w/ `_http_page_iter`?